docs/extensibility-workflow.md

```diff
@@ -0,0 +1,380 @@
+# Extensibility Workflow
+
+This guide covers advanced development workflows for extending StormWeaver, focusing on the action registry system and Lua C++ API extensions.
+
+## Action Registry System
+
+The action registry is at the core of StormWeaver's randomized testing framework.
+It manages a collection of weighted actions that are randomly selected and executed by worker threads.
+
+### Understanding the Action Architecture
+
+#### Core Components
+
+**Action Interface**
+```cpp
+class Action {
+public:
+  virtual void execute(metadata::Metadata &metaCtx, ps_random &rand,
+                       sql_variant::LoggedSQL *connection) const = 0;
+};
+```
+
+All actions must implement this interface with these key principles:
+- **Stateless**: Actions should be stateless to enable retry logic
+- **Thread-safe**: Multiple workers execute actions concurrently
+- **Metadata-aware**: Actions interact with shared database schema metadata
+
+Actions are expected to throw a `RuntimeException` (preferably `ActionException`) on failure.
+
+**ActionFactory Structure**
+```cpp
+struct ActionFactory {
+  std::string name;        // Unique identifier
+  action_build_t builder;  // Factory function
+  std::size_t weight;      // Probability weight
```

Comment:
would be nice to elaborate on weights a bit - accepted range, scale (e.g. `1000` has 10x more chances than `100`?)

Reply:
@dAdAbird I added a bit more details in the section above this, please check.